### PR TITLE
[READY] Highlander mode now includes silicons

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
I can't work on my tablet PR today, so,

- When Highlander mode triggers, all borgs will be desynced from their AI (if any), have their laws replaced with a law zero "THERE CAN ONLY BE ONE", and be force-set to the new highlander borg module. Due to the desync, all borgs are considered to be against eachother as well as against former crew.

- When Highlander mode triggers, all AIs will be disconnected from their shell (if any), be transformed into a borg, and have their old core gibbed. The above process will then be triggered as normal on their new borg mob.

- Adds a new highlander borg type, with two objects total: a highlander claymore and a nuke disk pinpointer.

- Highlander mode now uses a custom objective, child of the normal hijack, except does not allow any other borgs or animals of any type.

- Human Highlander contenders now receive a syndicate earpiece, so that they may engage in battle-appropriate banter even when telecomms is offline.

## Why It's Good For The Game
Borgs and AIs are no longer stuck with an impossible asimov lawset or (should the admin clear their laws) tempted to play favorites among the contenders. Instead they are contenders themselves.

## Changelog
:cl:
add: Cyborgs now have a Highlander module to use when said gamemode is enabled. The Highlander borg module comes equipped with a claymore, a pinpointer, and traditional garments for the occasion. Like humanoid contenders, Highlander borgs will die in three strikes.
add: AIs will become Highlander Cyborgs upon Highlander mode being triggered.
balance: Humanoid Highlander contenders now get a syndicate headset, so that banter may live on even if telecomms does not.
balance: Humanoid Highlander contenders are now immune to shocks, preventing them from being able to claim a cyborg killed them with an unfair shocked door.
fix: It is no longer possible to succeed as a highlander contenders if other contenders are alive on the shuttle.
fix: Borg tool storage huds no longer freak out if there are zero stored modules.
/:cl:

I wrote this code while I was stuck without access to my tablet PR. I do need to finish the borg module picker sprite, ~~and do a good round of testing on a proper computer.~~ New TODO list:

- [x] Fix borg claymore getting lost in the mail
- [x] Fix winning being possible with other highlanders on the shuttle
- [x] Fix some runtimes that show up in seemingly unrelated borg code that might actually be the cause of item 1

image

### CODE REVIEW

**Comment by GunHog** [comment]
What an adorable little murderer!

File: code/modules/antagonists/highlander/highlander.dm [hunk 1]
```diff
@@ -81,3 +81,23 @@
...
+	owner.special_role = "highlander"
+	give_equipment()
+
+/datum/antagonist/highlander/robot/give_equipment()
+	var/mob/living/silicon/robot/robotlander = owner.current
+	if(!istype(robotlander))
+		return ..()
+	robotlander.set_connected_ai() //disconnect from prior AI, if any
+	robotlander.laws.clear_inherent_laws()
+	robotlander.laws.set_zeroth_law("THERE CAN ONLY BE ONE")
```
Comment (Angustmeta) [inline code/modules/antagonists/highlander/highlander.dm:99]:
```suggestion
	robotlander.laws.set_zeroth_law("THERE CAN BE ONLY ONE")
```

**Comment by wesoda25** [comment]
... so no damage slowdown?

**Comment by itseasytosee** [comment]
highlanders don't normally have damage slowdown, I would adjust the borgs health to make it equal to your average Scotsman

**Comment by ATH1909** [comment]
> highlanders don't normally have damage slowdown, I would adjust the borgs health to make it equal to your average Scotsman

Yeah, currently, highborgs effectively have almost twice the health of a normal highlander.

Do highborgs get a syndicate encryption key? Can they hijack?

**Comment by zxaber** [comment]
I'll adjust the borg health when I get a chance in a bit.

Kiltborgs do get syndicomms (they are a child of syndiborgs), and the Highlander mode now has a special borg-compatible highjack.

**Comment by Barhandar** [comment]
Can average highlanders get sunglasses, unlike the borg?

**Comment by zxaber** [comment]
I don't think so. Far as I know, highlanders aren't allowed to pick anything up. But this borg module, being a special event-focused one, is the only module to not have a flash, so it shouldn't be a big deal either way.

**Comment by itseasytosee** [comment]
I would still argue that borgs have an advantage here in that they can manipulate doors to door things like shocking and bolt people down. They also have immunity to space.

**Comment by Doosteroni** [comment]
Yeah borgos being able to fuck with doors in kinda busted

**Comment by zxaber** [comment]
New changes:

- Highlander Borgs now have half the HP of a normal borg. This means that, like a human, they will die in three sword hits (technically the human only goes into crit after 3 hits, but I digress).

- Human(oid) Highlander contenders are now immune to shocks. This is mostly for show because everyone knows borgs are far too honorable to resort to shocking doors to cheese a cheap kill during the Highlander games.

- Bugfixing in general

Still need to do a full test with two clients tested.

**Review by Twaticus — APPROVED** [review]
ahaha

**Review by Timberpoes — CHANGES_REQUESTED** [review]
Please test the following possible edge case:

You have a code guard in place that prevents the cyborg modules from being broken.
Is there any way to repair the broken module 3?
Have you made sure said module can also never be repaired?

**Comment by Doosteroni** [comment]
uhh can they still bolt doors? Cause that's pretty fucking unfair if you are a normal human, since you don't get a fucking door bolt ability!

**Comment by zxaber** [comment]
>You have a code guard in place that prevents the cyborg modules from being broken.
Is there any way to repair the broken module 3?
Have you made sure said module can also never be repaired?

As per our chat in discord, it's really not a huge deal since the third slot being broken is basically fluff (the borg only has two tool items anyway). That said, now the slots won't repair if they're not designed to break.

**Comment by FloranOtten** [comment]
Can nukies buy one?

**Pull request merged**